scaleway/resource_instance_server.go

```diff
@@ -259,6 +271,12 @@ func resourceScalewayInstanceServerCreate(ctx context.Context, d *schema.Resourc
 		Tags:              expandStrings(d.Get("tags")),
 	}
 
+	if bootScriptID, ok := d.GetOk("bootscript_id"); ok {
+		b := instance.BootTypeBootscript
```

Comment:
IIRC a bootscript can be run in rescue mode too, that's why you should read `boot_type` attribute too